### PR TITLE
Update workflow config

### PR DESCRIPTION
**Pull request by alexvanyo**
This PR makes a few adjustments to the GitHub workflows:

- Fixes the `Release.yml` to be prepared for releasing APKs (this had project info copied from compose-samples)
- Simplifies the Gradle caching setup by using [gradle/gradle-build-action](https://github.com/gradle/gradle-build-action)
- Updates the reporting uploading to be flavor-specific

### CODE REVIEW

**Review by JolandaVerhoef — APPROVED** [review]

File: .github/workflows/Release.yml [hunk 1]
```diff
@@ -39,72 +39,12 @@ jobs:
...
+      - name: Upload app
         uses: actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: Crane/app/build/outputs/apk/debug/app-debug.apk
-          asset_name: crane-debug.apk
+          asset_path: app/build/outputs/apk/demo/debug/app-demo-debug.apk
+          asset_name: app-demo-debug.apk
```
Comment (JolandaVerhoef) [inline .github/workflows/Release.yml:49]: If we want to do a release this way, don't we want to create the release version of the app? APKs we want to distribute shouldn't have the performance implications of the debug build.

Comment (JoseAlcerreca) [reply .github/workflows/Release.yml:49]: +1, but how is the demoRelease apk signed?

Comment (JolandaVerhoef) [reply .github/workflows/Release.yml:49]: As discussed, it is currently signed with the debug keystore

Comment (alexvanyo) [reply .github/workflows/Release.yml]: Updated to release the demo release apk!

**Review by JolandaVerhoef — APPROVED** [review]

**Comment by JolandaVerhoef** [comment]
@alexvanyo this PR has conflicts, can you rebase?

**Comment by alexvanyo** [comment]
@JolandaVerhoef Done!

**Review by manuelvicnt — APPROVED** [review]

**Review by dturner — APPROVED** [review]

**Pull request merged**